tools/Logging/include/wil/result.h

```diff
@@ -3155,7 +3155,7 @@ inline void ReportFailure_Msg(
     ReportFailure(__R_FN_CALL_FULL, type, hr, message);
 }
 
-inline void ReportFailure_ReplaceMsg(__R_FN_PARAMS_FULL, FailureType type, HRESULT hr, _Printf_format_string_ PCSTR formatString, ...) {
+__declspec(noinline) inline void ReportFailure_ReplaceMsg(__R_FN_PARAMS_FULL, FailureType type, HRESULT hr, _Printf_format_string_ PCSTR formatString, ...) {
```

Comment:
What? "declspec(noinline) inline"???

Reply:
This is like `declspec(selectany)`, but for functions: It's not _inlined_, but it's also not emitted as a symbol in each translation unit. Info [here](https://blogs.msdn.microsoft.com/freik/2005/10/26/how-to-make-a-no-inline-inline-function-and-why-you-might-want-to/).